src/event-tracker.ts

```diff
@@ -32,85 +24,89 @@ const mixpanelMask = [
-				host: mixpanelHost.host,
-				path: mixpanelHost.path,
-			});
-		});
-	}
+	} = await config.getMany([
+		'unmanaged',
+		'mixpanelHost',
+		'mixpanelToken',
+		'uuid',
+	]);
+
+	defaultProperties = {
+		distinct_id: uuid,
+		uuid,
+		supervisorVersion,
+	};
 
-	public track(event: string, properties: EventTrackProperties | Error = {}) {
-		if (properties instanceof Error) {
-			properties = { error: properties };
-		}
-
-		properties = _.cloneDeep(properties);
-		if (properties.error instanceof Error) {
-			// Format the error for printing, to avoid display as { }
-			properties.error = {
-				message: properties.error.message,
-				stack: properties.error.stack,
-			};
-		}
-
-		// Don't send potentially sensitive information, by using a whitelist
-		properties = mask(properties, mixpanelMask);
-		this.logEvent('Event:', event, JSON.stringify(properties));
-		if (this.client == null) {
-			return;
-		}
-
-		properties = this.assignDefaultProperties(properties);
-		this.throttleddLogger(event)(properties);
+	if (unmanaged || mixpanelHost == null || mixpanelToken == null) {
+		return;
+	}
+	client = mixpanel.init(mixpanelToken, {
+		host: mixpanelHost.host,
+		path: mixpanelHost.path,
+	});
+})();
+
+export async function track(
+	event: string,
+	properties: EventTrackProperties | Error = {},
+) {
+	await initialized;
+
+	if (properties instanceof Error) {
+		properties = { error: properties };
 	}
 
-	private throttleddLogger = memoizee(
-		(event: string) => {
-			// Call this function at maximum once every minute
-			return _.throttle(
-				(properties: EventTrackProperties | Error) => {
-					this.client?.track(event, properties);
-				},
-				eventDebounceTime,
-				{ leading: true },
-			);
-		},
-		{ primitive: true },
-	);
-
-	private logEvent(...args: string[]) {
-		log.event(...args);
+	properties = _.cloneDeep(properties);
+	if (properties.error instanceof Error) {
+		// Format the error for printing, to avoid display as { }
+		properties.error = {
+			message: properties.error.message,
+			stack: properties.error.stack,
+		};
 	}
 
-	private assignDefaultProperties(
-		properties: EventTrackProperties,
-	): EventTrackProperties {
-		return _.merge({}, properties, this.defaultProperties);
+	// Don't send potentially sensitive information, by using a whitelist
+	properties = mask(properties, mixpanelMask);
+	logEvent('Event:', event, JSON.stringify(properties));
+	if (client == null) {
+		return;
 	}
+
+	properties = assignDefaultProperties(properties);
+	throttleddLogger(event)(properties);
 }
 
-export default EventTracker;
+const throttleddLogger = memoizee(
+	(event: string) => {
+		// Call this function at maximum once every minute
+		return _.throttle(
+			(properties: EventTrackProperties | Error) => {
+				client?.track(event, properties);
+			},
+			eventDebounceTime,
+			{ leading: true },
+		);
+	},
+	{ primitive: true },
+);
+
+export function logEvent(...args: string[]) {
+	log.event(...args);
+}
```

Comment:
This can now be changed to
```suggestion
export const logEvent = log.event.bind(log);
```
if you want

Reply:
Honestly I'm not sure why it's even still there, thinking about it.